lib/WebSocket.js

```diff
@@ -632,6 +634,14 @@ function initAsClient (address, protocols, options) {
 
   this._req = httpObj.get(requestOptions);
 
+  if (options.handshakeTimeout) {
+    this._req.setTimeout(options.handshakeTimeout, () => {
```

Comment:
This may not work exactly as you expect but it's fine in most cases. All `request.setTimeout()` does is calling `socket.setTimeout()` on the request socket.

This means that the timeout expires only when there are `x` milliseconds of inactivity. It's possible that the response arrives after the specified timeout with no 'timeout' event emitted.